src/sentry/integrations/metric_alerts.py

```diff
@@ -91,8 +100,125 @@ def incident_attachment_info(incident, new_status: IncidentStatus, metric_value=
     return {
         "title": title,
         "text": text,
-        "logo_url": logo_url,
+        "logo_url": LOGO_URL,
         "status": status,
         "ts": ts,
         "title_link": title_link,
     }
+
+
+def metric_alert_attachment_info(
+    alert_rule: AlertRule,
+    selected_incident: Optional[Incident] = None,
+    new_status: IncidentStatus = None,
+    metric_value=None,
+):
+    latest_incident = None
+    if selected_incident is None:
+        latest_incident = Incident.objects.filter(
+            id__in=Incident.objects.filter(alert_rule=alert_rule)
+            .values("alert_rule_id")
+            .annotate(incident_id=Max("id"))
+            .values("incident_id")
+        ).first()
+
+    if new_status:
+        status = INCIDENT_STATUS[new_status]
+    elif selected_incident:
+        status = INCIDENT_STATUS[IncidentStatus(selected_incident.status)]
+    elif latest_incident:
+        status = INCIDENT_STATUS[IncidentStatus(latest_incident.status)]
+    else:
+        status = INCIDENT_STATUS[IncidentStatus.CLOSED]
+
+    title = f"{status}: {alert_rule.name}"
+    title_link = absolute_uri(
+        reverse(
+            "sentry-metric-alert-details",
+            kwargs={
+                "organization_slug": alert_rule.organization.slug,
+                "alert_rule_id": alert_rule.id,
+            },
+        )
+    )
+    if selected_incident:
+        params = parse.urlencode({"alert": str(selected_incident.identifier)})
+        title_link += f"?{params}"
+
+    agg_display_key = alert_rule.snuba_query.aggregate
+
+    if CRASH_RATE_ALERT_AGGREGATE_ALIAS in alert_rule.snuba_query.aggregate:
+        agg_display_key = agg_display_key.split(f"AS {CRASH_RATE_ALERT_AGGREGATE_ALIAS}")[0].strip()
+
+    agg_text = QUERY_AGGREGATION_DISPLAY.get(agg_display_key, alert_rule.snuba_query.aggregate)
+
+    incident_info = selected_incident
+    if (
+        incident_info is None
+        and latest_incident
+        and latest_incident.status != IncidentStatus.CLOSED
+    ):
+        # Without selected incident, use latest incident if it is not resolved
+        incident_info = latest_incident
+
+    text = ""
```

Comment:
A lot of this logic looks like it's copied from `incident_attachment_info`. Can we try and DRY this up a bit?